common-lib/src/main/scala/com/gu/mediaservice/model/ThrallMessage.scala

```diff
@@ -1,85 +1,16 @@
 package com.gu.mediaservice.model
 
- import com.gu.mediaservice.lib.logging.{GridLogging, LogMarker}
- import com.gu.mediaservice.model.leases.MediaLease
- import com.gu.mediaservice.model.usage.UsageNotice
- import org.joda.time.{DateTime, DateTimeZone}
- import play.api.libs.json
- import play.api.libs.json.{JodaReads, JodaWrites, Json, OFormat, OWrites, Reads, __}
+import com.gu.mediaservice.lib.logging.{GridLogging, LogMarker}
 
-sealed trait ThrallMessage extends GridLogging with LogMarker {
-  val subject: String = this.getClass.getSimpleName()
-  val id: String
-  val lastModified: DateTime
-  val additionalMarkers: Map[String, Any] = Map()
-  def toJson = Json.toJson(this)
+trait ThrallMessage extends GridLogging with LogMarker {
+  def subject: () => String = () => this.getClass.getSimpleName
```

Comment:
Probably more a test of my Scala knowledge – why the change from `val` to `def` here?

Reply:
good catch, this was an artifact from Alex's changes, I've changed it back to val (and it doesn't need to be a function neither AFAIK) - will test it now